contracts/PBAB+Collabs/GenArt721CoreV2_IYKUpgradeable.sol

```diff
@@ -0,0 +1,867 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+// Created By: Art Blocks Inc.
+
+import "../interfaces/0.8.x/IRandomizer.sol";
+import "../interfaces/0.8.x/IGenArt721CoreV2_PBAB.sol";
+
+import "@openzeppelin-4.5/contracts/utils/cryptography/ECDSA.sol";
+import "@openzeppelin-4.5/contracts/utils/Strings.sol";
+import "@openzeppelin-4.5/contracts-upgradeable/utils/cryptography/ECDSAUpgradeable.sol";
+import "@openzeppelin-4.5/contracts-upgradeable/token/ERC721/ERC721Upgradeable.sol";
+import "@openzeppelin-4.5/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin-4.5/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+
+pragma solidity 0.8.9;
+
+/**
+ * @title Powered by Art Blocks ERC-721 core contract.
+ * @author Art Blocks Inc.
+ */
+contract GenArt721CoreV2_IYKUpgradeable is
```

Comment:
Is the diff between this file and their original contract (without the IYK stuff) simply the ERC721Upgradeable, UUPSUpgradeable, and Initializeable subclassing?

Reply:
Yup, exactly. It's a copy-paste of GenArt721CoreV2_PBAB, with:
* IYK functions/variables found in GenArt721CoreV2_IYK
* Subclassing the classes as you said
* With a initializer function & empty constructor
* With the `authorizeUpgrade` function implemented so the onlyAdmin modifier gates permission to perform the upgrade